ats/machines.py

```diff
@@ -317,8 +317,10 @@ def testEnded(self, test, status):
            #note test.status is not necessarily status after this!
            #see test.expectedResult
 
+        endNote = self.noteEnd(test)  #to be defined in children
+
         if not configuration.options.removeEndNote:
-            self.noteEnd(test)  #to be defined in children
+            print(endNote)
```

Comment:
Can you protect this with an 

if endNote:

so that if noteEnd does not return a string, nothing is printed?

Reply:
Yes, I can do that. Was also working on a print for ATS' flux module to be consistent with the other schedulers.